vendor/knative.dev/hack/codegen-library.sh

```diff
@@ -31,5 +31,5 @@ export GOBIN=${GOPATH}/bin # Set GOBIN explicitly as deepcopy-gen is installed b
 export CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${REPO_ROOT_DIR}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
 export KNATIVE_CODEGEN_PKG=${KNATIVE_CODEGEN_PKG:-$(cd ${REPO_ROOT_DIR}; ls -d -1 ./vendor/knative.dev/pkg 2>/dev/null || echo "${REPO_ROOT_DIR}")}
 
-chmod +x ${CODEGEN_PKG}/generate-groups.sh
-chmod +x ${KNATIVE_CODEGEN_PKG}/hack/generate-knative.sh
+[ -x ${CODEGEN_PKG}/generate-groups.sh ] || chmod +x ${CODEGEN_PKG}/generate-groups.sh
```

Comment:
Doing `chmod +x` might fail because CI pods do not run with root access and don't have permissions to change ownership.
Doing chmod will be done earlier when building the image: https://github.com/openshift-knative/hack/pull/80/files
So, the change you're asking about allows changing the ownership earlier and in this case "do nothing".

Reply:
Example:
```
podman run --user 1000 registry.ci.openshift.org/openshift/knative-eventing-src:knative-v1.11 chmod +x vendor/k8s.io/code-generator/generate-groups.sh
chmod: changing permissions of 'vendor/k8s.io/code-generator/generate-groups.sh': Operation not permitted
```

Reply:
When we do the chmod earlier in https://github.com/openshift-knative/hack/pull/80 in the image, why is it needed here?

Reply:
The `||` operator will make sure `chmod +x` will only be called if the file is not yet executable.
And we don't want to change the default behavior for upstream.

Reply:
> The || operator will make sure chmod +x will only be called if the file is not yet executable.

That's why I am wondering, as CI pods do not run with root access and don't have permissions to change it.
Maybe I only need the usecase, when this is relevant/needed here to understand it

Reply:
as discussed in the chat: :ok: